### PR TITLE
Fixed Cards that copy the Original name and effects

### PR DESCRIPTION
**Pull request by TheOnePharaoh**
Q.
スターヴ・ヴェノム・フュージョン・ドラゴン
相手が同じモンスターに（2）の効果を与えた場合。 ループを作成できますか？ または、1ターンに1度はまだ有効ですか？
A.
相手フィールドの「スターヴ・ヴェノム・フュージョン・ドラゴン」を対象に、「スターヴ・ヴェノム・フュージョン・ドラゴン」の『②』の効果を発動する事はできません。

Q.
Stave, Venom, Fusion, Dragon
When the opponent gives effect (2) to the same monster. Can I create a loop? Or is it still valid once per turn?
A.
You can not activate the effect of "②" of "Stave, Venom, Fusion, Dragon" for the opponent's field "Stave, Venom, Fusion, Dragon".

Ruling provided by Steeldarkeangel

Note:
Red Supremacy already applies that ruling and it's not a BKSS ruling because
the effect text says `.......this card's name becomes that target's original name, and .......`
the name can't become the target's original name if it already is with that same name
it's the same reason you can't target a monster with 0 ATK and (no negatable effect or if it's effects are already negated) using Clear Wing Fast Dragon
also the same reason you can't declare the current level in the effect of gagaga magician

### CODE REVIEW

**Comment by AndreOliveiraMendes** [comment]
i sugest you to revise some of these script cause you missed few details in some of they.

**Comment by Steeldarkeagel** [comment]
Multiple issues in this fix. Will close it down for now until I have been informed that these issues have been taken care of.

File: script/c25793414.lua [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	e2:SetProperty(EFFECT_FLAG_CARD_TARGET+EFFECT_FLAG_DELAY)
+	e2:SetCountLimit(1,25793415)
+	e2:SetCondition(c25793414.spcon)
+	e2:SetTarget(c25793414.sptg)
+	e2:SetOperation(c25793414.spop)
+	c:RegisterEffect(e2)
+end
+c25793414.miracle_synchro_fusion=true
+function c25793414.copyfilter(c,code)
+	return c:IsType(TYPE_MONSTER) and c:GetOriginalCode()~=code
```
Comment (TheOnePharaoh) [inline script/c25793414.lua:30]: this copies the original name
Crystal bell

File: script/c75620895.lua [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	e1:SetTarget(c75620895.target)
+	e1:SetOperation(c75620895.operation)
+	c:RegisterEffect(e1)
+end
+function c75620895.cost(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return e:GetHandler():CheckRemoveOverlayCard(tp,1,REASON_COST) end
+	e:GetHandler():RemoveOverlayCard(tp,1,1,REASON_COST)
+end
+function c75620895.filter(c,code)
+	return c:IsFaceup() and c:IsType(TYPE_XYZ) and c:GetCode()~=code
```
Comment (TheOnePharaoh) [inline script/c75620895.lua:24]: this copies the current name
One-Eyed skill gainer

File: script/c75620895.lua [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	local c=e:GetHandler()
+	local tc=Duel.GetFirstTarget()
+	if tc and c:IsRelateToEffect(e) and c:IsFaceup() and tc:IsFaceup() and tc:IsRelateToEffect(e) then
+		local code=tc:GetOriginalCode()
+		local e1=Effect.CreateEffect(c)
+		e1:SetType(EFFECT_TYPE_SINGLE)
+		e1:SetProperty(EFFECT_FLAG_CANNOT_DISABLE)
+		e1:SetReset(RESET_EVENT+0x1fe0000)
+		e1:SetCode(EFFECT_CHANGE_CODE)
+		e1:SetValue(tc:GetCode())
```
Comment (TheOnePharaoh) [inline script/c75620895.lua:42]: this was even copying the original code using the old script when it should've used the current code

**Review by TheOnePharaoh — COMMENTED** [review]
all the cards that should copy the current code use the Skill gainer check while the cards that copy the original name use the crystal bell check
but I'll review the effects that change the code again because in skill gainer the old script copied the original name when it should've copied the current name
I only fixed the effect in it because I noticed it in while testing of the card.

**Pull request closed without merging**